internal/peer/pubsub_test.go

```diff
@@ -0,0 +1,51 @@
+package peer
+
+import (
+	"testing"
+
+	"github.com/honeycombio/refinery/config"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_publicAddr(t *testing.T) {
+	cfg := &config.MockConfig{
+		GetPeerListenAddrVal:    "127.0.0.1:3443",
+		RedisIdentifier:         "somehostname",
+		IdentifierInterfaceName: "en0",
+	}
+	tests := []struct {
+		name    string
+		c       config.Config
+		want    string
+		wantErr bool
+	}{
+		{"basic", cfg, "http://somehostname:3443", false},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := publicAddr(tt.c)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("publicAddr() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != tt.want {
+				t.Errorf("publicAddr() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
```

Comment:
How about a test with multiple peers that can get correct number of peers and the callback functions are called when a peer joins or leaves the cluster?

Reply:
I've been trying to write that test, and it's not real easy to run multiple peers on the same machine. I *think* it's possible, but it might need some gymnastics to make it happen.

Because I'm going on vacation now, I'm going to merge this so it doesn't get stale while I'm gone. We can look at that test another time.